packages/desktop/components/popups/GetSeedPopup.svelte

```diff
@@ -0,0 +1,169 @@
+
+        if (!fileWithPath) {
+            return setFile()
+        }
+
+        const ext = /\.([0-9a-z]+)$/i.exec(fileWithPath.name)
+        if (!ext || !allowedExtensions.includes(ext[1])) {
+            return setFile()
+        }
+
+        const reader = new FileReader()
+
+        reader.onload = (e): void => {
+            setFile(e.target.result, fileWithPath)
+        }
+
+        reader.readAsArrayBuffer(fileWithPath)
+        reader.onloadend = (): void => {
+            extractSeedFromStrongholdFile()
+        }
+    }
+
+    async function extractSeedFromStrongholdFile(): Promise<void> {
+        await initialiseOnboardingProfile(false, true)
+        validateBackupFile(importFileName)
+        updateOnboardingProfile({
+            network: getDefaultPersistedNetwork(NetworkId.Iota),
+            importFile,
+            importFilePath,
+            // TODO: we don't have a way to know the stronghold version of the backup file yet
+            strongholdVersion: STRONGHOLD_VERSION,
+            type: ProfileType.Software,
+        })
+        await initialiseProfileManagerFromOnboardingProfile()
+        openUnlockStrongholdPopup()
+    }
+
+    async function getSeedFromSecretManager(): Promise<void> {
+        if (password) {
+            await restoreBackupFromStrongholdFile(password)
+        }
+        const managerId = await getProfileManager().id
+        const secretManager = await api.getSecretManager(managerId)
+        seed = await secretManager?.getSeed()
+    }
+
+    if (!readFromFile) {
+        getSeedFromSecretManager()
+    }
+</script>
+
+<div class="flex w-full flex-col space-y-6">
+    <Text type="h2">{title}</Text>
+    <div class="w-full p-10 bg-gray-50 dark:bg-gray-800 flex justify-center content-center">
+        <Logo width="50%" logo={LogoEnum.Stronghold} />
+    </div>
+    {#if seed}
+        <div class="flex flex-col space-y-4">
+            <div class="w-full space-y-3">
+                <Text type={TextType.p} color="gray-700">
+                    It is important to back up your seed to ensure you have a copy of your wallets and transaction
+                    history. If you lose your seed and recovery phrase you will lose access to your funds.
+                </Text>
+                <TextHint
+                    variant={TextHintVariant.Warning}
+                    text="You can import this seed into the new IOTA rebased wallet to access your funds."
+                />
+                <KeyValueBox keyText="Seed" valueText={seed} isCopyable />
+            </div>
+        </div>
+        <div class="flex flex-row justify-between space-x-4 w-full">
+            <Button outline classes="w-1/2" onClick={onCancelClick}>
+                {localize('actions.cancel')}
+            </Button>
+            <Button classes="w-1/2" onClick={onCopyClick}>
+                {localize('actions.copy')}
+            </Button>
+        </div>
+    {:else}
+        <div class="flex flex-col space-y-4">
```

Comment:
when using this option, if I enter the password wrong the popup just resets but doesnt show any error in the UI
![image](https://github.com/user-attachments/assets/30612b55-00f1-490b-b30f-619fff76b99d)
![image](https://github.com/user-attachments/assets/9e218c70-5cc7-4014-92bb-79767fcffc55)

Reply:
i think is because we need an loader to showcase that the unlocking of stronghold is trying to happen, when I enter the password correctly it also shows the popup empty for 1-2s

Reply:
https://github.com/user-attachments/assets/1462d93e-0c90-4a5a-8f6b-2e2b0c3efa44